app/controllers/link_check_reports_controller.rb

```diff
@@ -6,14 +6,53 @@ def create
       section_id: link_reportable_params[:section_id]
     )
 
-    service.call
+    @report = service.call
 
-    head :created
+    @reportable = find_reportable(link_reportable_params)
+
+    respond_to do |format|
+      format.js { render 'admin/link_check_reports/create' }
+      format.html { redirect_to_reportable }
+    end
+  end
+
+  def show
+    @report = LinkCheckReport::ShowService.new(
+      id: link_reportable_show_params[:id]
+    ).call
+
+    @reportable = find_reportable(section_id: @report.section_id, manual_id: @report.manual_id)
+
+    respond_to do |format|
+      format.js { render 'admin/link_check_reports/show' }
+      format.html { redirect_to_reportable }
+    end
   end
 
 private
 
   def link_reportable_params
     params.require(:link_reportable).permit(:manual_id, :section_id)
   end
+
+  def link_reportable_show_params
+    params.permit(:id)
+  end
+
+  def find_reportable(reportable_params)
```

Comment:
We already have something similar to this method in the `LinkCheckReport` class, are we able to reuse that at all?

Reply:
I think we might something similar in the Whitehall version 🤔 ... The model here is [pretty barebones](https://github.com/alphagov/manuals-publisher/blob/master/app/models/link_check_report.rb).

Reply:
Ah... I was thinking of the code here: https://github.com/alphagov/manuals-publisher/blob/1380109a7e1bcb5c49d49fb40a3c933e66119fb3/app/services/link_check_report/create_service.rb#L51

I wonder if we can extract them out into something else perhaps?

Reply:
@thomasleese this change went in with e122429